.clang-format

```diff
@@ -0,0 +1,194 @@
+BreakBeforeTernaryOperators: true
+# Kt N/A. The closest is https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting AfterColon looks the most consistent
+BreakConstructorInitializers: AfterColon
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting
+BreakInheritanceList: AfterColon
+# Kt U. Choosing to break long strings to fit into line length
+BreakStringLiterals: true
+# Kt U. IDEA displays a vertical guide at 120. Choosing it to avoid very long lines.
+ColumnLimit: 140
+# Kt N/A. Probably should use nested namespaces instead.
+CompactNamespaces: false
+# Kt N/A. The closest is https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#formatting
+ConstructorInitializerIndentWidth: 4
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#formatting IDEA seems to indent continuations with 8
+ContinuationIndentWidth: 8
+# Kt N/A. The closest is https://kotlinlang.org/docs/reference/coding-conventions.html#horizontal-whitespace
+# Braced initalizer braces are close to parens in function call or to brackets in array initialier. And styleguide asks not to put spaces in them.
+Cpp11BracedListStyle: true
+# Configured by PointerAlignment
+DerivePointerAlignment: false
+# Confgured by BinPack*
+ExperimentalAutoDetectBinPacking: false
+# Kt N/A. Setting to true because it helps to see when anonymous namespace ends
+FixNamespaceComments: true
+# Kt U. Not touching manually created including blocks
+IncludeBlocks: Preserve
+# Kt U. Sorting: main header (priority 0) > system headers > project headers
+IncludeCategories:
+  - Regex: '^<.*'
+    Priority: 1
+  - Regex: '.*'
+    Priority: 2
+# Kt N/A. Main header must match current filename (modulo extension) exactly.
+IncludeIsMainRegex: '$'
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#formatting-control-flow-statements
+IndentCaseLabels: true
+# Kt N/A. Do not indent macro code
+IndentPPDirectives: None
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#formatting
+IndentWidth: 4
+# Kt U. IDEA seems to indent
+IndentWrappedFunctionNames: true
+# Kt U. Do not touch.
+KeepEmptyLinesAtTheStartOfBlocks: false
+# Kt U. IDEA keeps 1 empty line by default.
+MaxEmptyLinesToKeep: 1
+# Kt N/A.
+NamespaceIndentation: None
+# Kt N/A. The closest is https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting
+ObjCBinPackProtocolList: Never
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#formatting
+ObjCBlockIndentWidth: 4
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#annotation-formatting
+ObjCSpaceAfterProperty: false
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#horizontal-whitespace
+ObjCSpaceBeforeProtocolList: false
```

Comment:
Protocols are not like type parameters.
Is it
```obj-c
@interface Foo <Bar>
```
vs
```obj-c
@interface Foo<Bar>
```
?
How do popular style guides handle this?

Reply:
For this comment and the one below it:
```
> clang-format -style=LLVM -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: false
ObjCSpaceBeforeProtocolList: true
> clang-format -style=Google -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: false
ObjCSpaceBeforeProtocolList: true
> clang-format -style=Mozilla -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: true
ObjCSpaceBeforeProtocolList: false
> clang-format -style=WebKit -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: true
ObjCSpaceBeforeProtocolList: true
```